### PR TITLE
Recombination

### PR DESCRIPTION
**Pull request by andrewkern**
got the recombination code implemented, training loss working but not val loss! @nspope take a look and see if you can catch something

### CODE REVIEW

**Comment by nspope** [comment]
tracked down the issue-- it was a bug introduced to `process_ts_batch.py` in the vcf prediction PR

File: workflow/scripts/ts_simulators.py [hunk 1]
```diff
@@ -267,11 +270,12 @@ class recombination_rate(BaseSimulator):
         "pop_size": 1e4,
         # RANDOM PARAMETERS (UNIFORM)
         "recombination_rate": [0, 1e-8],
+        "dummy": [0, 1e-8],
```
Comment (andrewkern) [inline workflow/scripts/ts_simulators.py:273]: what's up with this dummy random param?

Comment (nspope) [reply workflow/scripts/ts_simulators.py:273]: that's me debugging; i'm not done checking stuff out

**Comment by nspope** [comment]
This seems to be working fine for a separately trained embedding network, but not end-to-end (which doesn't learn at all). I'm going to try to figure out what's going on ...

**Comment by andrewkern** [comment]
huh. have you tried turning the shuffling off on end-to-end? i wonder if that could somehow screw things up?

**Comment by nspope** [comment]
i'm seeing the same issue with the vanilla RNN on the AraTha example -- separate training works fine, end to end not so much. I think I'm going to go ahead and merge this to get the bigger fix in, and then troubleshoot the end-to-end pipeline.
